extensions/amp-analytics/0.1/variables.js

```diff
@@ -141,6 +141,34 @@ function replaceMacro(string, matchPattern, opt_newSubStr) {
   return string.replace(regex, opt_newSubStr);
 }
 
+/**
+ * Applies the match function to the given string with the given regex
+ * @param {string} string input to be replaced
+ * @param {string} matchPattern string representation of regex pattern
+ * @param {string=} opt_indexStr the matching group to return. Index of 0
+ *                  indicates the full match. Defaults to 0
+ * @return {string} returns the matching group given by opt_indexStr
+ */
+function matchMacro(string, matchPattern, opt_indexStr) {
+  if (!matchPattern) {
+    user().warn(TAG, 'MATCH macro must have two or more arguments');
+  }
+
+  let index = 0;
+  if (opt_indexStr) {
+    index = parseInt(opt_indexStr, 10);
+  }
+
+  // if given a non-number or negative number
+  if (!index || index < 0) {
```

Comment:
nit: It'd be great if we can throw a user error if index is invalid. (NaN or negative)